inference-engine/thirdparty/clDNN/api/cldnn/primitives/gather_nd.hpp

```diff
@@ -19,25 +19,37 @@ struct gather_nd : public primitive_base<gather_nd> {
     CLDNN_DECLARE_PRIMITIVE(gather_nd)
 
     /// @brief Constructs gather_nd primitive.
-    /// @param id This primitive id.
-    /// @param data Input data primitive id.
-    /// @param indices Input indexes primitive id.
-    /// @param indices_rank Rank of indices.
-    /// @param batch_dims batch_dims as an attribute of GatherND. Optional.
+    ///
+    /// @param id                   This primitive id.
+    /// @param data                 Input data primitive id.
+    /// @param indices              Input indexes primitive id.
+    /// @param indices_rank         Rank of indices.
+    /// @param batch_dims           batch_dims as an attribute of GatherND. Optional.
+    /// @param batch_merged_output  batched output shape is merged as a dimention for v5.
+    ///                             In case of output{3, 2, 4, 5} at batch_dims = 2, real output shape should be {6, 4, 5}.
+    ///                             This should be false for v8.
+    ///                             For batch_dims < 2, This doesn't have any meaning.
     gather_nd(const primitive_id& id,
               const primitive_id& data,
               const primitive_id& indices,
               const uint8_t indices_rank,
               const uint8_t batch_dims = 0,
+              const bool batch_merged_output = true,
```

Comment:
I think the newly added argument are better to be located at the last position for the backward comppatibility

Reply:
Your suggestion is reasonable. However, almost all other primitives have the params, 'ext_prim_id' and 'output_padding' at the last position. I think this consistency is more important.